### PR TITLE
Honor prefix parameter in ListBucketPolicies API

### LINKED ISSUE #926: ListBucketPolicies does not filter policy rules using objectPrefix parameter
When reading the code of ListBucketPolicies() (https://github.com/minio/minio-go/blob/master/api-get-policy.go#L52), we can see that objectPrefix is not being used to filter policy rules although ListBucketPolicies() claims to do so.

```
// ListBucketPolicies - list all policies for a given prefix and all its children.
func (c Client) ListBucketPolicies(bucketName, objectPrefix string) (bucketPolicies map[string]policy.BucketPolicy, err error) {
        // Input validation.
        if err := s3utils.CheckValidBucketName(bucketName); err != nil {
                return map[string]policy.BucketPolicy{}, err
        }
        if err := s3utils.CheckValidObjectNamePrefix(objectPrefix); err != nil {
                return map[string]policy.BucketPolicy{}, err
        }
        policyInfo, err := c.getBucketPolicy(bucketName)
        if err != nil {
                errResponse := ToErrorResponse(err)
                if errResponse.Code == "NoSuchBucketPolicy" {
                        return map[string]policy.BucketPolicy{}, nil
                }       
                return map[string]policy.BucketPolicy{}, err
        }
        return policy.GetPolicies(policyInfo.Statements, bucketName), nil
}
```
This needs to be fixed.

### PR DESCRIPTION
**Pull request by kannappanr**
Fixes #926

### CODE REVIEW

File: pkg/policy/bucket-policy.go [hunk 1]
```diff
@@ -620,6 +620,33 @@ func GetPolicies(statements []Statement, bucketName string) map[string]BucketPol
 	return policyRules
 }
 
+// GetPrefixPolicies - returns a map of policies of given bucket name, prefix in given statements.
+func GetPrefixPolicies(statements []Statement, bucketName, prefix string) map[string]BucketPolicy {
```
Comment (harshavardhana) [inline pkg/policy/bucket-policy.go]: actually a better function name for go style is `GetPoliciesWithPrefix` helps in editor auto completion and also conveys the purpose.

Comment (kannappanr) [reply pkg/policy/bucket-policy.go]: done

**Comment by kannappanr** [comment]
@harshavardhana Changes made. PTAL

File: pkg/policy/bucket-policy.go [hunk 2]
```diff
@@ -620,6 +620,33 @@ func GetPolicies(statements []Statement, bucketName string) map[string]BucketPol
 	return policyRules
 }
 
+// GetPoliciesWithPrefix - returns a map of policies of given bucket name, prefix in given statements.
+func GetPoliciesWithPrefix(statements []Statement, bucketName, prefix string) map[string]BucketPolicy {
```
Comment (vadmeste) [inline pkg/policy/bucket-policy.go]: It looks like there is little difference between GetPolicies & GetPoliciesWithPrefix, they look practically the same, can you do with this style ?

```
func GetPolicies(statements []Statement, bucketName string) map[string]BucketPolicy {
   return GetPoliciesWithPrefix(statements, bucketName, "")
}
```

Comment (kannappanr) [reply pkg/policy/bucket-policy.go]: Handling both the cases in GetPolicies function itself.

**Pull request merged**